processor/eventfilter/eventfilter.go

```
@@ -69,3 +77,325 @@ func GetSupportedMessageEvents(destination *backendconfig.DestinationT) ([]strin
+				},
+				"hybrid": {
+					"messageType": { "allowedValues": ["group","track"] },
+				}
+			},
+			"web": {
+				"cloud": {
+					"messageType": { "allowedValues": ["group","track"] },
+				},
+				"device": {
+					"messageType": { "allowAll": true },
+				}
+			}
+		},
+		...
+	}
+*/
+func FilterUsingSupportedConnectionModes(connectionModeFilterParams ConnectionModeFilterParams) (bool, *transformer.TransformerResponseT) {
+	destination := connectionModeFilterParams.Destination
+	srcType := connectionModeFilterParams.SrcType
+	messageType := connectionModeFilterParams.Event.MessageType
+	evaluatedDefaultBehaviour := connectionModeFilterParams.DefaultBehaviour
+
+	/*
+		Cases:
+		1. if supportedConnectionModes is not present -- rely on default behaviour
+		2. if sourceType not in supportedConnectionModes && sourceType is in supportedSourceTypes (the sourceType is not defined) -- rely on default behaviour
+		3. if sourceType = {} -- don't send the event(silently drop the event)
+		4. if connectionMode not in supportedConnectionModes.sourceType (some connectionMode is not defined) -- don't send the event(silently drop the event)
+		5. if sourceType.connectionMode = {} -- rely on default behaviour
+		6. "device" is defined as a key for a sourceType -- don't send the event
+		7. if eventProperty not in sourceType.connectionMode -- rely on property based default behaviour
+		8. if sourceType.connectionMode.eventProperty = {} (both allowAll, allowedValues are not included in defConfig) -- rely on default behaviour
+		9. if sourceType.connectionMode.eventProperty = { allowAll: true, allowedValues: ["track", "page"]} (both allowAll, allowedValues are included in defConfig) -- rely on default behaviour
+	*/
+
+	destConnModeI := misc.MapLookup(destination.Config, "connectionMode")
+	if destConnModeI == nil {
+		pkgLogger.Debug("connectionMode not present, filtering event based on default behaviour")
+		return evaluatedDefaultBehaviour, nil
+	}
```

Instead of ignoring the error, we can use for better log message, plus is more idiomatic to check for absent error rather than nil

```suggestion
	destConnModeI, err := misc.MapLookup(destination.Config, "connectionMode")
	if err != nil {
		pkgLogger.Debug("connectionMode not present, filtering event based on default behaviour: %v", err)
		return evaluatedDefaultBehaviour, nil
	}
```

---

`MapLookup` returns an `interface{}` only. Hence the check condition was written as such.